app/views/spaces_new.erb

```diff
@@ -1,31 +1,31 @@
 <% @title="List a Space | MakersBnB" %>
 
-<h2>List a Space</h2>
+<h2><%= @space ? "Update Space" : "List a Space" %></h2>
 <form action="/spaces" method="post">
   <label for="name">
     Name:
-    <input type="text" name="name" required>
+    <input type="text" name="name" value="<%= @space.name if @space %>" required>
   </label>
 
   <label for="description">
     Description:
-    <textarea name="description" cols="30" rows="10"></textarea>
+    <textarea name="description" cols="30" rows="10"><%= @space.description if @space %></textarea>
   </label>
 
   <label for='price'>
      Price:
-    <input type='number' name='price' step="0.01" required/>
+    <input type='number' name='price' step="0.01" value="<%= @space.price if @space %>" required/>
```

Comment:
Why not just keep space_new.erb like it was before and then just use spaces_update.erb for the /spaces/update path, instead of having condititions everywhere?